Remotion/Data/DomainObjects/Persistence/Rdbms/Parameters/SimpleDataParameterDefinitionFactory.cs

```diff
@@ -0,0 +1,47 @@
+// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using Remotion.Data.DomainObjects.Persistence.Rdbms.Model.Building;
+using Remotion.Utilities;
+
+namespace Remotion.Data.DomainObjects.Persistence.Rdbms.Parameters;
+
+/// <summary>
+/// Can create <see cref="IDataParameterDefinition"/> instances that can handle simple (atomic) parameter values.
+/// </summary>
+public class SimpleDataParameterDefinitionFactory
```

Comment:
I mirrored the design of `DataStoragePropertyDefinitionFactory` (minus the interfaces for the partial factory types and the `ObjectIDWithoutClassID` implementation).
Idea: make `SimpleDataParameterDefinitionFactory` and `ObjectIDDataParameterDefinitionFactory` implement a new interface `ISingleValueDataParameterDefinitionFactory`, inject a collection of `ISingleValueDataParameterDefinitionFactory` into `DataStoragePropertyDefinitionFactory` and use them in a chain of responsibility in `CreateStoragePropertyDefinition`.

Reply:
Ah, yes. Hmm... further consideration: make just IDataStoragePropertyDefinitionFactory types and the chain of responsibility as decorators. so, each type will either handle the QueryParameter or pass it on. 

Having two interfaces, once for orchestration of the chain and one as the chain items feels overkill for the design? Only advantage is that the chain would be implemented as an iteration instead of a pseduo-recursive call. 

I like that we could simply use a decorator chain here. Unless it's ugly? thoughts...?

Reply:
As we talked about, this is going to be redesigned into a decorator-style chain of responsibility.